taskchampion/src/taskdb/apply.rs

```diff
@@ -2,6 +2,44 @@ use crate::errors::{Error, Result};
 use crate::operation::Operation;
 use crate::server::SyncOp;
 use crate::storage::{StorageTxn, TaskMap};
+use crate::Operations;
+
+/// Apply `operations` to the dataabase, in the given single transaction.
+///
+/// This updates the set of tasks in the database, but does not modify the list of operations.
+///
+/// The transaction is not committed.
+pub(super) fn apply_operations(txn: &mut dyn StorageTxn, operations: &Operations) -> Result<()> {
+    for operation in operations {
+        match operation {
+            Operation::Create { uuid } => {
+                txn.create_task(*uuid)?;
+            }
+            Operation::Delete { uuid, .. } => {
+                txn.delete_task(*uuid)?;
+            }
+            Operation::Update {
+                uuid,
+                property,
+                value,
+                ..
+            } => {
+                // TODO: could cache this value from op to op
+                let task = txn.get_task(*uuid)?;
+                if let Some(mut task) = task {
```

Comment:
So an Update operation is ignored if the uuid isn't already in the database, right? Shouldn't that be an error or at least some kind of warning?

(edit: I see apply_operations_update_does_not_exist tests for this behavior below.)

Reply:
I'll add a docstring.